File: ruby/ruby_bandwidth_bxml/verbs/forward.rb
require_relative 'xml_verb'

module RubyBandwidthBxml
  class Forward
    include XmlVerb

    def to_xml(xml)
      xml.Forward(compact_hash({
       'to' => to,
       'from' => from,
       'callTimeout' => call_timeout,
       'diversionTreatment' => diversion_treatment,
       'diversionReason' => diversion_reason
      }))
    end
  end
end

File: csharp/BandwidthBXML/src/BandwidthBXML/verbs/Forward.cs
﻿using System.ComponentModel;
using System.Xml.Serialization;

namespace BandwidthBXML
{
  /// <summary>
  ///   The Forward verb is used to transfer the call to another number.
  /// </summary>
  public class Forward : IVerb
  {
    /// <summary>
    /// The phone number destination of the call
    /// </summary>
    [XmlAttribute("to")]
    public string To { get; set; }

    /// <summary>
    /// The phone number that will make the forward call
    /// </summary>
    [XmlAttribute("from")]
    public string From { get; set; }

    /// <summary>
    /// The number of seconds to wait before timing out the call
    /// </summary>
    [XmlAttribute("callTimeout")]
    public int CallTimeout { get; set; }

    /// <summary>
    /// Diversion treatment for the transfer
    /// </summary>
    [XmlAttribute("diversionTreatment")]
    public string DiversionTreatment { get; set; }

    /// <summary>
    /// Diversion reason for the transfer
    /// </summary>
    [XmlAttribute("diversionReason")]
    public string DiversionReason { get; set; }
  }
}
